ui/packages/app/src/Index.svelte

```diff
@@ -122,21 +122,15 @@
 
 	function handle_darkmode(target: HTMLDivElement): "light" | "dark" {
 		let url = new URL(window.location.toString());
-		let theme: "light" | "dark" = "light";
-
-		const color_mode: "light" | "dark" | "system" | null = url.searchParams.get(
-			"__theme"
-		) as "light" | "dark" | "system" | null;
-
-		if (color_mode !== null) {
-			if (color_mode === "dark") {
-				theme = darkmode(target);
-			} else if (color_mode === "system") {
-				theme = use_system_theme(target);
-			}
-			// light is default, so we don't need to do anything else
-		} else if (url.searchParams.get("__dark-theme") === "true") {
-			theme = darkmode(target);
+		let url_color_mode: "light" | "dark" | "system" | null =
+			url.searchParams.get("__theme") as "light" | "dark" | "system" | null;
+		url_color_mode =
+			url_color_mode ||
+			(url.searchParams.get("__dark-theme") === "true" ? "dark" : null);
+		theme = url_color_mode || theme;
+
+		if (theme === "dark" || theme === "light") {
+			darkmode(target, theme);
```

Comment:
Can this be refactored to remove the shorthand's and ternaries? This is far more difficult to reason about than the previous block of if statements.

Reply:
The user provided theme value should take precedence here as well.user mode > query param > system. I think we can get rid of them `__dark-theme` thing now but will update when @julien-c or @gary149 get back to me.

Also we could create a type for `type ThemeVariant =  "light" | "dark" | "system"` for ease of maintenance / readability.

So if we leave `user_theme_variant` undefined if it isn't passed, i think this can all be simplified to:

```ts
let theme_variant_query_param = url.searchParams.get("__theme") as ThemeVariant || undefined
active_theme_variant: ThemeVariant || undefined = user_theme_variant || theme_variant_query_param

if (theme_variant) {
  darkmode(target, theme_variant);
} else {
  active_theme_variant = use_system_theme(target);
}
```